Objective-C/CBLDatabase.mm

```diff
@@ -1460,7 +1251,7 @@ - (BOOL) saveResolvedDocument: (CBLDocument*)resolvedDoc
         }
         
         if (resolvedDoc != localDoc)
-            resolvedDoc.database = self;
+            resolvedDoc.collection = [self defaultCollectionOrThrow];
```

Comment:
Same as `- (bool)resolveConflictInDocument:withConflictResolver:error`, this `saveResolvedDocument:withLocalDoc:remoteDoc:error:` should be moved to collection class.